client/lib/analytics/ad-tracking/record-order.js

```diff
@@ -432,6 +437,61 @@ function recordOrderInGoogleAds( cart, orderId ) {
 	}
 }
 
+function recordOrderInGAEnhancedEcommerce( cart, orderId, wpcomJetpackCartInfo ) {
+	if ( ! isAdTrackingAllowed() ) {
+		debug( 'recordOrderInGAEnhancedEcommerce: [Skipping] ad tracking is disallowed' );
+		return;
+	}
+
+	if ( ! isGoogleAnalyticsEnabled || ! isGoogleAnalyticsEnhancedEcommerceEnabled ) {
+		debug( 'recordOrderInGAEnhancedEcommerce: [Skipping] Google Analytics is not enabled' );
+		return;
+	}
+
+	let products, brand, totalCost;
+
+	if ( wpcomJetpackCartInfo.containsWpcomProducts ) {
+		products = wpcomJetpackCartInfo.wpcomProducts;
+		brand = GA_PRODUCT_BRAND_WPCOM;
+		totalCost = wpcomJetpackCartInfo.wpcomCost;
+	} else if ( wpcomJetpackCartInfo.containsJetpackProducts ) {
+		products = wpcomJetpackCartInfo.jetpackProducts;
+		brand = GA_PRODUCT_BRAND_JETPACK;
+		totalCost = wpcomJetpackCartInfo.jetpackCost;
+	} else {
+		debug( 'recordOrderInGAEnhancedEcommerce: [Skipping] No products' );
+		return;
+	}
+
+	const items = [];
+	products.map( product => {
+		items.push( {
+			id: product.product_id,
```

Comment:
Similarly, is this already a string or Number? Should be a string.

Reply:
I've changed this to be more explicit with the data type.